core/types/transaction_signing.go

```diff
@@ -133,7 +134,7 @@ func (s EIP155Signer) Sender(tx InternalTransaction) (common.Address, error) {
 		return HomesteadSigner{}.Sender(tx)
 	}
 
-	if tx.ChainID().Cmp(s.chainID) != 0 {
+	if tx.ChainID().Cmp(nodeconfig.GetDefaultConfig().GetNetworkType().ChainConfig().EthCompatibleChainID) == -1 && tx.ChainID().Cmp(s.chainID) != 0 {
```

Comment:
for ethCompatibleChainID, it should also check != 0. Since we know exactly what's the right ethChainID for this shard.